src/input_device/JoystickPool.h

```diff
@@ -59,17 +59,16 @@ struct JoystickAction : public ObjectCounter<JoystickAction>
 
 	explicit JoystickAction(std::string string);
 	JoystickAction(int _joyid, Type _type, int _number)
-		: type(_type), joy(nullptr), joyid(_joyid),
-			number(_number) {}
+		: type(_type)
```

Comment:
just to make sure I understand:
the source of the crash was that we were storing an `SDL_Joystick` pointer, which would be invalid after plug-out/plug-in, whereas now we are only storing the ID so the `nullptr` check will catch a missing joystick?

Reply:
Yes, this was exactly the problem. I edited the description be using your summary.